### PR TITLE
[GH-157] - correct handling of CallInfo<T> usages

### LINKED ISSUE #157: Handle CallInfo<T> from upcoming NSubstitute version
More info https://github.com/nsubstitute/NSubstitute/pull/641#issuecomment-768629653

### PR DESCRIPTION
**Pull request by tpodolak**
Closes #157

### CODE REVIEW

**Comment by tomasz-podolak** [comment]
@dtchepak `libs/nsubstitute-latest/NSubstitute.dll` - this one is built from https://github.com/nsubstitute/NSubstitute/pull/641, once it is released I will update this file

**Comment by tpodolak** [comment]
@ `libs/nsubstitute-latest/NSubstitute.dll` this one is built from https://github.com/nsubstitute/NSubstitute/pull/641. Once new version of NSubstitute is released, I will update `libs` folder

File: tests/NSubstitute.Analyzers.Tests.Shared/NSubstituteVersion.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace NSubstitute.Analyzers.Tests.Shared
+{
+    public enum NSubstituteVersion
+    {
+        NSubstitute4_2_2 = 1,
```
Comment (tpodolak) [inline tests/NSubstitute.Analyzers.Tests.Shared/NSubstituteVersion.cs:5]: I am following  similar naming convention as compiler team does in https://github.com/dotnet/roslyn/blob/d0d9a59a9c222e58e679645ff5fdc2c074c8335e/src/Compilers/CSharp/Portable/LanguageVersion.cs#L12
It doesnt look too elegant IMO but I didnt figure out anything better. Any other options?

Comment (dtchepak) [reply tests/NSubstitute.Analyzers.Tests.Shared/NSubstituteVersion.cs:5]: I think this is ok 👍

**Review by dtchepak — APPROVED** [review]

**Pull request merged**